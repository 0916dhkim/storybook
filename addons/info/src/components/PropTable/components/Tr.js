import PropTypes from 'prop-types';
import React from 'react';

const Tr = ({ children }) => <tr>{children}</tr>;

Tr.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.arrayOf(PropTypes.element)])
    .isRequired,
};

export default Tr;
