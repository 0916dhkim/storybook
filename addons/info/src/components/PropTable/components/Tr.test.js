import React from 'react';
import { shallow } from 'enzyme';

import Tr from './Tr';

describe('PropTable/Tr', () => {
  describe('renders a tr html node with react element children', () => {
    const wrapper = shallow(
      <Tr>
        <div>foo bar</div>
        <div>baz</div>
      </Tr>
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('renders a tr html node with html node children', () => {
    const wrapper = shallow(<Tr>foo bar baz</Tr>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('renders a tr html node with one child node', () => {
    const wrapper = shallow(<Tr>foo bar</Tr>);
    expect(wrapper).toMatchSnapshot();
  });
});
