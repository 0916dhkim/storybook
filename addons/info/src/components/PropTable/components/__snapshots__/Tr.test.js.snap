// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<tr>
  <div>
    foo bar
  </div>
  <div>
    baz
  </div>
</tr>
`;

exports[` 2`] = `
<tr>
  foo bar
</tr>
`;

exports[` 3`] = `
<tr>
  foo
  bar
</tr>
`;
